web-app/js/portal/prototypes/Ext.js

```diff
@@ -34,5 +34,16 @@ Ext.tree.TreeNode.prototype.clone = function(recursive) {
     }
     
     return clone;
-}
+};
 
+Ext.tree.TreeNode.prototype.getHierarchy = function(attribute) {
```

Comment:
Cool with me. Mind if we invite a couple of others to take a look at before final decision?

Pretty sure @dnahodil and @jkburges will have an opinion. Guys?